xbmc/filesystem/NFSFile.cpp

```diff
@@ -45,16 +45,16 @@ using namespace std::chrono_literals;
 
 namespace
 {
+// Default "lease_time" on most Linux NFSv4 servers are 90s.
+// See: https://linux-nfs.org/wiki/index.php/NFS_lock_recovery_notes
+// Keep alive interval should be always less than lease_time to avoid client session expires
 
-constexpr auto CONTEXT_TIMEOUT = 6min;
-
-constexpr auto KEEP_ALIVE_TIMEOUT = 3min;
-
-constexpr auto IDLE_TIMEOUT = 3min;
+constexpr auto CONTEXT_TIMEOUT = 60s; // 2/3 parts of lease_time
```

Comment:
Why do these other values need to be adjusted?

Reply:
Because context "expires" when time of inactivity is greater than server's "lease_time"

Reply:
Current solution is:

CONTEXT_TIMEOUT = 6 minutes
But expires in 90s then when someone opens a file after 90s nfs_open fails with NFS4ERR_EXPIRED but is retry with https://github.com/xbmc/xbmc/commit/9dc30e83c830d379c063eaf3a4a64a9fb57ca782


New solution is:

Discard expired contexts so `nfs_open` should never fail and should never retry...